@labath, on Arm I'm seeing these 2 failures:

```
Failed Tests (2):
  lldb-api :: functionalities/breakpoint/delayed_breakpoints/TestDelayedBreakpoint.py
  lldb-api :: functionalities/thread/crash_during_step/TestCrashDuringStep.py
```

```
******************** TEST 'lldb-api :: functionalities/thread/crash_during_step/TestCrashDuringStep.py' FAILED ********************
Script:
--

UNSUPPORTED: LLDB (/home/leandro.lupori/git/ci/build32/bin/clang-arm) :: test_step_inst_with_dsym (TestCrashDuringStep.CrashDuringStepTestCase) (test case does not fall in any category of interest for this run)
FAIL: LLDB (/home/leandro.lupori/git/ci/build32/bin/clang-arm) :: test_step_inst_with_dwarf (TestCrashDuringStep.CrashDuringStepTestCase)
Log Files:
 - /home/leandro.lupori/git/ci/build32/lldb-test-build.noindex/functionalities/thread/crash_during_step/TestCrashDuringStep/Failure_test_step_inst_with_dwarf.log
FAIL: LLDB (/home/leandro.lupori/git/ci/build32/bin/clang-arm) :: test_step_inst_with_dwo (TestCrashDuringStep.CrashDuringStepTestCase)
Log Files:
 - /home/leandro.lupori/git/ci/build32/lldb-test-build.noindex/functionalities/thread/crash_during_step/TestCrashDuringStep/Failure_test_step_inst_with_dwo.log
======================================================================
FAIL: test_step_inst_with_dwarf (TestCrashDuringStep.CrashDuringStepTestCase)
   Test thread creation during step-inst handling.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/leandro.lupori/git/ci/llvm-project/lldb/packages/Python/lldbsuite/test/lldbtest.py", line 2042, in test_method
    return attrvalue(self)
  File "/home/leandro.lupori/git/ci/llvm-project/lldb/packages/Python/lldbsuite/test/decorators.py", line 160, in wrapper
    return func(*args, **kwargs)
  File "/home/leandro.lupori/git/ci/llvm-project/lldb/test/API/functionalities/thread/crash_during_step/TestCrashDuringStep.py", line 48, in test_step_inst_with
    self.assertEqual(process.GetState(), lldb.eStateStopped, PROCESS_STOPPED)
AssertionError: 10 != 5 : Process status should be stopped
Config=arm-/home/leandro.lupori/git/ci/build32/bin/clang
======================================================================
FAIL: test_step_inst_with_dwo (TestCrashDuringStep.CrashDuringStepTestCase)
   Test thread creation during step-inst handling.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/leandro.lupori/git/ci/llvm-project/lldb/packages/Python/lldbsuite/test/lldbtest.py", line 2042, in test_method
    return attrvalue(self)
  File "/home/leandro.lupori/git/ci/llvm-project/lldb/packages/Python/lldbsuite/test/decorators.py", line 160, in wrapper
    return func(*args, **kwargs)
  File "/home/leandro.lupori/git/ci/llvm-project/lldb/test/API/functionalities/thread/crash_during_step/TestCrashDuringStep.py", line 48, in test_step_inst_with
    self.assertEqual(process.GetState(), lldb.eStateStopped, PROCESS_STOPPED)
AssertionError: 10 != 5 : Process status should be stopped
Config=arm-/home/leandro.lupori/git/ci/build32/bin/clang
----------------------------------------------------------------------
Ran 3 tests in 0.560s

FAILED (failures=2, skipped=1)

--

********************
```

```
******************** TEST 'lldb-api :: functionalities/breakpoint/delayed_breakpoints/TestDelayedBreakpoint.py' FAILED ********************
Script:
--
/usr/bin/python3.10 /home/leandro.lupori/git/ci/llvm-project/lldb/test/API/dotest.py -u CXXFLAGS -u CFLAGS --env LLVM_LIBS_DIR=/home/leandro.lupori/git/ci/build32/./lib --env LLVM_INCLUDE_DIR=/home/leandro.lupori/git/ci/build32/include --env LLVM_TOOLS_DIR=/home/leandro.lupori/git/ci/build32/./bin --triple armv8l-unknown-linux-gnueabihf --build-dir /home/leandro.lupori/git/ci/build32/lldb-test-build.noindex --lldb-module-cache-dir /home/leandro.lupori/git/ci/build32/lldb-test-build.noindex/module-cache-lldb/lldb-api --clang-module-cache-dir /home/leandro.lupori/git/ci/build32/lldb-test-build.noindex/module-cache-clang/lldb-api --executable /home/leandro.lupori/git/ci/build32/./bin/lldb --compiler /home/leandro.lupori/git/ci/build32/./bin/clang --dsymutil /home/leandro.lupori/git/ci/build32/./bin/dsymutil --make /usr/bin/gmake --llvm-tools-dir /home/leandro.lupori/git/ci/build32/./bin --lldb-obj-root /home/leandro.lupori/git/ci/build32/tools/lldb --lldb-libs-dir /home/leandro.lupori/git/ci/build32/./lib --cmake-build-type Release /home/leandro.lupori/git/ci/llvm-project/lldb/test/API/functionalities/breakpoint/delayed_breakpoints -p TestDelayedBreakpoint.py
--
Exit Code: 1

Command Output (stdout):
--
lldb version 23.0.0git (git@github.com:luporl/llvm-project.git revision af6732a77058dc6c275667b651a82b2d3b6b61d4)
  clang revision af6732a77058dc6c275667b651a82b2d3b6b61d4
  llvm revision af6732a77058dc6c275667b651a82b2d3b6b61d4
Skipping the following test categories: libc++, msvcstl, dsym, pdb, gmodules, debugserver, objc

--
Command Output (stderr):
--
UNSUPPORTED: LLDB (/home/leandro.lupori/git/ci/build32/bin/clang-arm) :: test_dsym (TestDelayedBreakpoint.TestDelayedBreakpoint) (test case does not fall in any category of interest for this run)
PASS: LLDB (/home/leandro.lupori/git/ci/build32/bin/clang-arm) :: test_dwarf (TestDelayedBreakpoint.TestDelayedBreakpoint)
PASS: LLDB (/home/leandro.lupori/git/ci/build32/bin/clang-arm) :: test_dwo (TestDelayedBreakpoint.TestDelayedBreakpoint)
UNSUPPORTED: LLDB (/home/leandro.lupori/git/ci/build32/bin/clang-arm) :: test_eager_breakpoints_dsym (TestDelayedBreakpoint.TestDelayedBreakpoint) (test case does not fall in any category of interest for this run)
warning: failed to set breakpoint site at 0x1234566 for breakpoint 3.1: error sending the breakpoint request: 9
FAIL: LLDB (/home/leandro.lupori/git/ci/build32/bin/clang-arm) :: test_eager_breakpoints_dwarf (TestDelayedBreakpoint.TestDelayedBreakpoint)
Log Files:

FAIL: test_eager_breakpoints_dwarf (TestDelayedBreakpoint.TestDelayedBreakpoint)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/leandro.lupori/git/ci/llvm-project/lldb/packages/Python/lldbsuite/test/lldbtest.py", line 2042, in test_method
    return attrvalue(self)
  File "/home/leandro.lupori/git/ci/llvm-project/lldb/test/API/functionalities/breakpoint/delayed_breakpoints/TestDelayedBreakpoint.py", line 79, in test_eager_breakpoints
    self.assertIn(addr, breakpoint_lines)
AssertionError: '1234567' not found in 'python3.10       <  16> send packet: $Z0,1234566,2#7fpython3.10       <  62> send packet: $jMultiBreakpoint:{"breakpoint_requests":["Z0,3710618,4"]}]#17'
Config=arm-/home/leandro.lupori/git/ci/build32/bin/clang
======================================================================
FAIL: test_eager_breakpoints_dwo (TestDelayedBreakpoint.TestDelayedBreakpoint)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/leandro.lupori/git/ci/llvm-project/lldb/packages/Python/lldbsuite/test/lldbtest.py", line 2042, in test_method
    return attrvalue(self)
  File "/home/leandro.lupori/git/ci/llvm-project/lldb/test/API/functionalities/breakpoint/delayed_breakpoints/TestDelayedBreakpoint.py", line 79, in test_eager_breakpoints
    self.assertIn(addr, breakpoint_lines)
AssertionError: '1234567' not found in 'python3.10       <  16> send packet: $Z0,1234566,2#7fpython3.10       <  62> send packet: $jMultiBreakpoint:{"breakpoint_requests":["Z0,98d0618,4"]}]#51'
Config=arm-/home/leandro.lupori/git/ci/build32/bin/clang
----------------------------------------------------------------------
Ran 6 tests in 1.051s

FAILED (failures=2, skipped=2)

--

********************
```

`TestDelayedBreakpoint.py` was fixed #196888, so it seems  `functionalities/thread/crash_during_step/TestCrashDuringStep.py` is the only failure remaining with this patch.

--- a/lldb/source/Plugins/Process/Utility/NativeProcessSoftwareSingleStep.cpp
+++ b/lldb/source/Plugins/Process/Utility/NativeProcessSoftwareSingleStep.cpp
@@ -127,8 +127,10 @@ Status NativeProcessSoftwareSingleStep::SetupSoftwareSingleStepping(
 
     // If setting the breakpoint fails because pc is out of the address
     // space, ignore it and let the debugee segfault.
-    if (error.GetError() == EIO || error.GetError() == EFAULT)
+    if (error.GetError() == EIO || error.GetError() == EFAULT) {
+      error.Clear();
       continue;
+    }
     if (error.Fail())
       return error;
 

